spec/models/mixins/drift_state_mixin_spec.rb

```diff
@@ -0,0 +1,96 @@
+describe DriftStateMixin do
+  include Spec::Support::ArelHelper
+
+  let(:host) { FactoryGirl.create(:host) }
+
+  let(:drift_states) do
+    [
+      FactoryGirl.create(:drift_state, :resource => host, :timestamp => recent_timestamp, :data => "bogus"),
+      FactoryGirl.create(:drift_state, :resource => host, :timestamp => old_timestamp, :data => "bogus")
+    ]
+  end
+
+  let(:recent_timestamp) { 2.months.ago.change(:usec => 0) }
+  let(:old_timestamp) { 4.months.ago.change(:usec => 0) }
+
+  describe "#first_drift_state" do
+    it "uses the most recent value" do
+      drift_states
+      expect(host.last_drift_state.timestamp).to eq(recent_timestamp)
+    end
+  end
+
+  describe "#last_drift_state" do
+    it "uses the least recent value" do
+      drift_states
+      expect(host.first_drift_state.timestamp).to eq(old_timestamp)
+    end
+  end
+
+  describe "#last_drift_state_timestamp" do
+    describe "with no timestamps" do
```

Comment:
I realize `describe` and `context` are functionally equivalent in rspec, but can you use `context` instead of `describe` for each of the blocks for the different `timestamp` states?

Reply:
and the other places too. thnx